### PR TITLE
[IMP] web,account,stock,sale & others: adapt report layouts to new report editor

### PR DESCRIPTION
**Pull request by bouvyd**
General guidelines for adaptations:
- include 'demo data' inside the report (e.g. `<span t-field='o.name'>INV/2023/001</span>`)
- include div.oe_structure elements in several places - these elements take no space in the printed report if empty, but allow users to 'hook in' several places of the reports
- modify the base report layout to include several placeholder blocks (for company address, etc.) that make it clear what goes where (since demo data are a bit unclear for bigger blocks, like address blocks)
- add branching or re-order some t-if/t-else conditional blocks. The report shows the first `t-if` content by default, so it makes sense to have it be 'the default value' (e.g. instead of showing the options table all the time in the editor for a quote report, only show is as a t-else) - the goal is to have the report editor open on a report that looks as much as possible as the default, vanilla version of the report
- whenever possible, try to avoid `<t>` nodes insise `table/thead/tbody/tr` nodes, as this is not allowed by the hml spec and browser parser will not agree with those. Since the problem is known, the report editor will convert such tables to div elements (and preserve the layout as much as possible) - but making sure the table is 'correct html' (e.g. by moving qweb directives on legal nodes, like tr or td) will preserve the layout and will also allow users to use the table modification tool of the Odd Editor
This is a first pass on 'important' reports. A second pass will come before v17.

### CODE REVIEW

**Comment by bouvyd** [comment]
@robodoo rebase-ff

**Comment by bouvyd** [comment]
@robodoo rebase-merge

**Comment by bouvyd** [comment]
Headsup: I will **not** wait for a review by all invovled people according to @C3POdoo and the codeowner system. This will get merged soon and is mostly harmless. No need to have half the gurus review it.

**Review by xavierbol — APPROVED** [review]

File: addons/product_matrix/views/matrix_templates.xml
Comment (Feyensv) [inline addons/product_matrix/views/matrix_templates.xml:32]: over indented :eyes:

**Comment by bouvyd** [comment]
@robodoo r+

**Pull request closed without merging**
